File: python/easy/calc_chars.py
def calc_chars(s):
  word = ''.join(sorted(s.strip()))
  hash = {}
  result = ''

  for char in word:
    if char in hash:
      hash[char] = hash[char] + 1
    else:
      hash[char] = 1

  for char in hash:
    result += f'{char}{hash[char]}'
  return result


print(calc_chars('BBBAADDDDDECCCC'))

File: ts/easy/calcChars.ts
export const calcChars = (str: string): string => {
  const s = str.trim().split('').sort().join('');
  const hashTable: Record<string, number> = {};
  let result = '';

  for (let i = 0; i < s.length; i++) {
    hashTable[s[i]] = hashTable[s[i]] ? hashTable[s[i]] + 1 : 1;
  }

  for (const [key, value] of Object.entries(hashTable)) {
    result += key + '' + value;
  }
  return result;
};
console.log(calcChars('BBBAADDDDDECCCC'));
